gravitee-am-gateway/gravitee-am-gateway-handler/gravitee-am-gateway-handler-common/src/main/java/io/gravitee/am/gateway/handler/common/factor/FactorManager.java

```diff
@@ -30,4 +33,6 @@ public interface FactorManager extends Service {
     Factor getFactor(String factorId);
 
     void updateFactor(String factorId);
+
+    Optional<Factor> getActiveFactor(Client client, String factorId);
```

Comment:
`getActiveFactor` may be misleading as it could also mean the end user as already enrolled  and challenge ? 

How about `getClientFactor` ?

Reply:
FP